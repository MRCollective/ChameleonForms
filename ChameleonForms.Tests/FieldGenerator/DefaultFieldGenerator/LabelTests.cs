using ApprovalTests.Html;
using ChameleonForms.Component.Config;
using NUnit.Framework;

namespace ChameleonForms.Tests.FieldGenerator.DefaultFieldGenerator
{
    class LabelTests : DefaultFieldGeneratorShould
    {
        [Test]
        public void Use_correct_html_for_field_label_without_config()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(null);

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_label_with_null_label()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(new FieldConfiguration());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_label_with_overridden_label()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(new FieldConfiguration().Label("asdf"));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_label_with_overridden_id()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(new FieldConfiguration().Attr(id => "DifferentId"));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_label_with_overridden_id_2()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(new FieldConfiguration().Id("DifferentId"));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_validation()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetValidationHtml(null);

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }
    }
}
