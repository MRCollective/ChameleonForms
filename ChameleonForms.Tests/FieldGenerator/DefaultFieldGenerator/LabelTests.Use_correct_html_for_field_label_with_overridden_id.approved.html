<label for="DifferentId">RequiredString</label>